### ISSUE TITLE
[Bug]: Chromium returns "en-US@posix" for `navigator.language`

### ISSUE THREAD

**Issue by Torgen** [microsoft/playwright#34046]
### Version

1.49.0 and 1.49.1

### Steps to reproduce

in node repl:
```js
const { chromium } = require('playwright')
await browser = chromium.launch()
await page = browser.newPage()
await page.evaluate("navigator.language")
```

### Expected behavior

It should return any legal BCP47 language code.

### Actual behavior

'en-US@posix'

### Additional context

Running on Ubuntu 24.04 on WSL2, but something similarly illegal is getting returned on my github actions runners based on how my tests are failing  in https://github.com/Torgen/codex-blackboard/pull/1361 and https://github.com/Torgen/codex-blackboard/pull/1374. Basically it's consistent with https://github.com/Torgen/codex-blackboard/blob/master/client/imports/timestamp.js throwing an error when I try to pass `navigator.language` to the constructor of `Intl.DateTimeFormat` at the module top level. I don't expect you to run those, but they're how I noticed there was a problem.

### Environment

```shell
System:
    OS: Linux 5.4 Ubuntu 24.04 LTS 24.04 LTS (Noble Numbat)
    CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
    Memory: 7.06 GB / 7.73 GB
    Container: Yes
  Binaries:
    Node: 18.19.1 - /usr/bin/node
    Yarn: 1.22.22 - /usr/local/bin/yarn
    npm: 9.2.0 - /usr/bin/npm
  IDEs:
    VSCode: 1.95.3 - /mnt/c/Users/Dan/AppData/Local/Programs/Microsoft VS Code/bin/code
  Languages:
    Bash: 5.2.21 - /usr/bin/bash
  npmPackages:
    @playwright/browser-chromium: ^1.49.1 => 1.49.1
    @playwright/browser-firefox: ^1.49.1 => 1.49.1
    playwright: ^1.49.1 => 1.49.1
```

**Comment by pavelfeldman** [comment]
https://github.com/adobe/react-spectrum/issues/7457 has a good analysis and links. Sounds like overriding language in the config is the only workaround.

**Comment by krassowski** [comment]
This has bitten Jupyter ecosystem as well, breaking CI on dozens of repositories. If my understanding is correct, this would be a bug in Chromium and not in playwright, correct?

I see that the Chromium team stumbled upon this in May and committed a workaround for their use case in https://issues.chromium.org/issues/337047877.

**Comment by Torgen** [comment]
Tried setting LANG=en_US on both my node repl example and on my actual tests and it seemed to be an effective workaround.

**Comment by mxschmitt** [comment]
I was not able to reproduce on WSL2 with Ubuntu 24.04 on Windows 11. @Torgen what was the original value of echo $LANG? or did `env` yield something interesting?

**Comment by Torgen** [comment]
```shell
$ echo $LANG
C.UTF-8
```

**Comment by pavelfeldman** [comment]
@krassowski 

> If my understanding is correct, this would be a bug in Chromium and not in playwright, correct?

Yes, and the workaround with setting the LANG env variable above sounds like the easiest fix. It is a little hard to file the issue upstream https://issues.chromium.org/issues?q=status:open without the repro. If you have the repro, could you file it and mention it here? The issue you point to only mentions it briefly, so it was not addressed in full.

**Comment by dgozman** [comment]
Closing as per above. If you file a chromium issue, please post a link here as well.

**Comment by uhyo** [comment]
Hi everyone 👋 let me share that setting `LANG=en_US` didn't work for us (as of `playwright@1.15.1`). Instead, `LC_ALL=en_US` worked.

You can test this using the Docker image:

```sh
docker run -it mcr.microsoft.com/playwright:v1.51.1 bash
```

Then inside your Docker container:

```shell
root@49b6fef13c60:/proj# npm init --yes
root@49b6fef13c60:/proj# npm install playwright-core
...
root@49b6fef13c60:/proj# cat > main.mjs
import { chromium } from 'playwright-core';
const browser = await chromium.launch();
const page = await browser.newPage();
const result = await page.evaluate("navigator.language");
console.log(result);
await browser.close();
root@49b6fef13c60:/proj# node main.mjs
en-US@posix
root@49b6fef13c60:/proj# LANG=en_US node main.mjs
en-US@posix
root@49b6fef13c60:/proj# LC_ALL=en_US node main.mjs
en-US
```

**Comment by gregtws** [comment]
For others an alternate workaround is mentioned in this bug https://github.com/chromedp/chromedp/issues/922#issuecomment-937500493   Basically you need to copy (or symlink) the locales directory into the headless shell directory as its missing.

e.g.

```
$ cd ~/.cache/ms-playwright/chromium_headless_shell-XXXX/chrome-linux
$ ln -s ../../chromium-XXXX/chrome-linux/locales locales
```

Note this bug also impacts navigator in service workers as well (the same env var or symlink workaround works for both regular navigator and that in service workers).  There is a separate issue in that `navigator.languages` does not report correctly (e.g. If it should be reporting `en-US,en`, it reports `en-US`)  This may be  related to an emulation bug in Chromium -- https://issues.chromium.org/issues/40236995)
